ocsci/pytest_customization/ocscilib.py

```diff
@@ -103,3 +103,13 @@ def process_cluster_cli_params(config):
         cluster_path = os.path.join(cluster_dir_parent, cluster_name)
     ocsci_config.ENV_DATA['cluster_name'] = cluster_name
     ocsci_config.ENV_DATA['cluster_path'] = cluster_path
+
+
+def pytest_collection_modifyitems(session, config, items):
+    """
+    Add Polarion ID property to test cases that are marked with one.
+    """
+    for item in items:
+        for marker in item.iter_markers(name="polarion_id"):
```

Comment:
I would rather use https://docs.pytest.org/en/latest/reference.html#_pytest.nodes.Node.get_closest_marker WDYT?

Reply:
Reiterating what we discussed earlier: I think `get_closest_marker` is a good idea. This could save us in the event that both a test method and the test class it resides in are marked with polarion_ids that conflict. With `get_closest_marker` we would only use the value from the test level marker and not the class. This is still a situation we don't want to have present in our tests, but we also don't want to post results to the wrong tests in the event it happens. +1 for `get_closest_marker`.

Reply:
Done

Reply:
What if one automated test covers more test cases in Polarion? (not sure, if it is "allowed" or not, but I think, it is possible to have such case...)

On the other side it doesn't make sense to add Polarion ID on class level, because it doesn't make sense to have more automated tests reporting to one test case in Polarion...

Reply:
@dahorak I'm not sure if the importer will be able to handle this anyway.